CMakeLists.txt

```diff
@@ -126,6 +126,15 @@ option(Scylla_CHECK_HEADERS
   "Add check-headers target for checking the self-containness of headers")
 if(Scylla_CHECK_HEADERS)
     add_custom_target(check-headers)
+    # compatibility target used by CI, which builds "dev-headers" for building
+    # "headers" of the "dev" mode. but we don't have such an alias. what we
+    # have is check-headers:Dev, and we don't want to build check-headers
+    # for all "CMAKE_DEFAULT_CONFIGS". so, before we , let's add an target
+    # which only build build check-headers for Dev, and skip it otherwise.
+    add_custom_target(dev-headers
+        COMMAND ${CMAKE_COMMAND}
+          "$<IF:$<CONFIG:Dev>,--build;${CMAKE_BINARY_DIR};--config;$<CONFIG>;--target;check-headers,-E;echo;skipping;dev-headers;in;$<CONFIG>>"
+        COMMAND_EXPAND_LISTS)
 endif()
```

Comment:
Let's push this in first. I want to see comparisons of compiler and linker flags before we merge it.

Perhaps a way to compare is to split the command line to a word per line, sort, and compare the sorted results. It doesn't have to be exactly equal, but I want to understand the differences.

Reply:
i will create a script so one can repeat the verification before doing the switch.